great_expectations/datasource/fluent/interfaces.py

```diff
@@ -508,7 +517,7 @@ def parse_batching_regex_string(
     ) -> re.Pattern:
         pattern: re.Pattern
         if not batching_regex:
-            pattern = re.compile(".*")
+            pattern = MATCH_ALL_PATTERN
```

Comment:
❤️

Reply:
Can we move `MATCH_ALL_PATTERN` to the default argument and remove this if branch? That is `def parse_batching_regex_string(batching_regex: Union[re.Pattern, str] = MATCH_ALL_PATTERN)`.

Reply:
Yes we can! 😄 
But some of the other `Datasource` types rely on it. Once they've all been updated with the pattern in this PR we can remove it.